pontoon/checks/libraries/comparelocales.py

```diff
@@ -0,0 +1,166 @@
+from collections import (
+    namedtuple,
+)
+
+from compare_locales.checks import getChecker
+from compare_locales.parser import (
+    FluentParser,
+)
+
+CommentEntity = namedtuple(
+    'Comment', (
+        'all',
+    )
+)
+
+File = namedtuple(
+    'File', (
+        'file',
+        'locale',
+    )
+)
+
+# Because we can't pass the context to all entities passed to compare locales,
+# we have to create our equivalents of compare-locale's internal classes.
+ComparePropertiesEntity = namedtuple(
+    'ComparePropertiesEntity',
+    (
+        'key',
+        'val',
+        'raw_val',
+        'pre_comment',
+    )
+)
+CompareDTDEntity = namedtuple(
+    'CompareDTDEntity',
+    (
+        'key',
+        'val',
+        'raw_val',
+        'pre_comment',
+        'all',
+    )
+)
+
+
+DTD_ENTITY_TMPL = '<!ENTITY %s \"%s\">'
+
+
+class UnsupportedResourceTypeError(Exception):
+    """Raise if compare-locales doesn't support given resource-type."""
+    pass
+
+
+def cast_to_compare_locales(resource_ext, entity, string):
+    """
+    Cast a Pontoon's translation object into Entities supported by `compare-locales`.
+
+    :arg basestring resource_ext: extension of a resource.
+    :arg pontoon.base.models.Entity entity: Source entity
+    :arg basestring string: a translation
+    :arg pontoon.base.models.Locale locale: Locale of a translation
+    :return: source entity and translation entity that will be compatible a compare-locales checker.
+        Type of those entities depends on the resource_ext.
+    """
+    if resource_ext == '.properties':
+        return (
+            ComparePropertiesEntity(
+                entity.key,
+                entity.string,
+                entity.string,
+                CommentEntity(entity.comment)
+            ),
+            ComparePropertiesEntity(
+                entity.key,
+                string,
+                string,
+                CommentEntity(entity.comment),
+            )
+        )
+
+    elif resource_ext == '.dtd':
+        return (
+            CompareDTDEntity(
+                entity.key,
+                entity.string,
+                entity.string,
+                CommentEntity(entity.comment),
+                DTD_ENTITY_TMPL % (entity.key, entity.string)
+            ),
+            CompareDTDEntity(
+                entity.key,
+                string,
+                string,
+                CommentEntity(entity.comment),
+                DTD_ENTITY_TMPL % (entity.key, entity.string)
+            )
+        )
+
+    elif resource_ext == '.ftl':
+        parser = FluentParser()
+
+        parser.readContents(entity.string)
```

Comment:
`readContents` takes bytes, you'll need to entity.string.encode('utf-8') here. For now, that is.  https://bugzilla.mozilla.org/show_bug.cgi?id=1448256 might add a dedicated unicode entry point.